docs/cookbook/production-deployment.md

```diff
@@ -0,0 +1,308 @@
+```
+docker save production-php-fpm | gzip | ssh -oStrictHostKeyChecking=no -i <PRIVATE_KEY_PATH> root@<YOUR_DOMAIN_HERE> 'gunzip | docker load'
+```
+
+### Deployment On Production Server
+
+#### First Setup Deploy
+
+Now we need to copy [`docker-compose-prod-deploy.yml.dist`](../../docker/conf/docker-compose-prod-deploy.yml.dist) into folder on the production server as `docker-compose.yml`.
+After the image is in the repository of the production server we create docker containers and run phing target `build-new`,
+to build application for production with clean DB and base data.
+We use parameter `-p` to specify the name of the project and prefix for the volumes so these will be easily accessible.
+There are named volumes created under path `/var/lib/docker/volumes/` and one persisted folder `production-content` in `PATH_TO_DIRECTORY_WHERE_IS_DOCKER_COMPOSE_YML_STORED` for all uploaded images and generated files that should not be removed.  
+We create persited folder with correct owner and symlink to web folder for static files
+```
+cd PATH_TO_DIRECTORY_WHERE_IS_DOCKER_COMPOSE_YML_STORED #(e.g. /var/www/html)
+mkdir /var/www/production-content
+chown -R $(id -un): /var/www/production-content
+ln -s /var/lib/docker/volumes/production_web-volume/_data/ web
+```
+and start containers with docker-compose.
+```
+docker-compose -p production up -d
+docker-compose -p production exec php-fpm ./phing build-new
+```
+
+Now the docker container should run.
+We want to setup scheduler for execution of cron jobs by adding one line into `/etc/crontab` file.
+Cron job is executed every 5 minutes in `php-fpm` container.
+```
+*/5 * * * * /usr/bin/docker exec production-php-fpm php phing cron
+```
+Since web application is running we can go to administration and change passwords for default administrators.
+With login `superadmin` and password `admin123` we can do it via these urls:
+* `/admin/administrator/edit/1`
+* `/admin/administrator/edit/2`
+
+Now we go to `/admin/dashboard/` and fulfill all requests that are demanding for us by red colored links.
+
+#### Next Deploys
+
+To preserve created data we need to use phing target `build` that consists of two commands 
+* `build-deploy-part-1-db-independent` no maintenance page is needed during execution of this command 
+* `build-deploy-part-2-db-dependent` maintenance page is needed if there are exist unapplied database migrations  
+
+With each update of master branch in our repository we need to rebuild image based on [Docker Image Building](#docker_image_building) section.
+
+We are logged in production server via ssh and start to deploy new built production image.
+```
+cd PATH_TO_DIRECTORY_WHERE_IS_DOCKER_COMPOSE_YML_STORED (e.g. /var/www/html)
+
+# make sure that container for building the application image is cleared
+docker rm -f build-php-fpm-container
+
+# launch container for building the application image
+docker run --detach --rm --name build-php-fpm-container \
```

Comment:
Wait, this container gets deleted right after run ends with `--rm` i think. Also you just run this container and do not run anything in it right? So it just starts and gets deleted?

Reply:
this container should start the same way like `docker-compose -p production up php-fpm` with default command `docker-php-entrypoint php-fpm` it is running the whole time and at the end it is stopped and removed by `docker rm -f build-php-fpm-container`

Reply:
--rm deleted